### DISCUSSION TITLE
Coder template w/ terraform modules and private git repo

### DISCUSSION THREAD

**Discussion by fcma-sheigl** [coder/coder#15137] [General]
Hi, I am trying to use modules within a template. The modules are in our on prem ADO using a private git repository. I can't figure out how to add authentication headers using an ADO personal access token to the internal coder git command. I have added the auth headers to my local git and can clone fine there, but with `coder templates push` on the terraform build, it errors out because it can't authenticate. Below is an example of the module:

```terraform
module "a_module" {
  source   = "git::https://localado/acollection/aproject/_git/Coder.Terraform.Modules//modulename"
  agent_id = coder_agent.main.id
  install_version = jsondecode(data.coder_parameter.dotnet_version.value)
}
```

and the way I added my PAT to my local git is:

```bash
AUTH_HEADER="AUTHORIZATION: Basic $B64_PAT"
ADO_BASEURL=https://localado/acollection
APPDEV_GIT=$ADO_BASEURL/aproject/_git

git config --global --add http.$APPDEV_GIT/.extraHeader "$AUTH_HEADER"
```

**Comment by matifali** [comment]
Hi @fcma-sheigl, you should be able to use a token to fetch from the private git repo as described here: https://coder.com/docs/admin/templates/extending-templates/modules#private-git-repository.

Let me know if you need further help.
